text/0086-sentry-bundler-plugins-api.md

```diff
@@ -0,0 +1,103 @@
+  // If not provided, this option will default to its default values.
+  release?: {
+    name?: string; // if not provided - Sentry CLI will try to guess a name (ie. git sha, cloud provider env vars, etc.)
+    inject?: boolean; // default: true - whether to inject the release into bundles
+    create?: boolean; // default: true - if set to `false` will not create a release in Sentry, no matter what other options are set - the release value will still be injected though.
+    finalize?: boolean; // default: true
+    dist?: string; // Can be used to segment releases further
+    vcsRemote?: string; // default: 'origin' - Version control system remote name
+    setCommits?: SetCommitsOptions;
+    deploy?: DeployOptions;
+    legacySourcemaps?: {
+      // Pre-debugid way of uploading source maps - for users that for whatever reason cannot inject code into their bundles
+      include: string | IncludeEntry | Array<string | IncludeEntry>;
+      cleanArtifacts?: boolean; // default: false - Wether to delete artifacts previously uploaded to the release
+    };
+  };
+
+  _experiments: {}; // whatever
+}
+
+// Identical to status-quo. See https://github.com/getsentry/sentry-webpack-plugin/blob/2b7d274a7355f0d27a431b2c20c37c9786bbe4cb/README.md for more information.
+interface IncludeEntry {
+  // ...
+}
+
+// Identical to status-quo. See https://github.com/getsentry/sentry-webpack-plugin/blob/2b7d274a7355f0d27a431b2c20c37c9786bbe4cb/README.md for more information.
+interface SetCommitsOptions {
+  // ...
+}
+```
+
+Some removed options:
+
+- `dryRun` - Instead, just not use the plugin.
+- `configFile` - The config file is very intransparent in combination with the plugin and generally not well documented. Let's remove this feature to lower maintenance burden.
+- `injectReleasesMap` - Was used to support some micro frontends setups. This is not needed anymore with the introduction of debug IDs.
+- `releaseInjectionTargets` - This was used to disable/enable release injection for a particular set of files. This is not really useful and cannot be used in all bundlers (esbuild). Additionally, for technical reasons we need to inject the release into every processed module anyhow (it will be deduped) so this option simply doesn't make a lot of sense anymore.
+
+# Drawbacks
+
+- Users will have to jump through a few hoops upgrading to the new Webpack plugin major just based on changing the options type.
+- Users may have to adjust their code AND build system if we drop environment variables as a way of configuring the plugin.
+- We are maintaining two systems of uploading source maps.
+
+# Unresolved questions
+
+- Should the options we intend to remove really be removed or do we see use cases for them?
+- Should we remove environment variables and `.sentryclirc` as a means to configure the plugin?
+- Should we deprecate the legacy source maps upload system entirely?
```

Comment:
This is too large of a breaking change IMO - let's just keep it around, like we do with `extras`.